sql/core/src/test/resources/sql-tests/inputs/trunc.sql

```diff
@@ -0,0 +1,136 @@
+-- trunc decimal
```

Comment:
Can you add some tests for the result type, specially important for DECIMAL()

Reply:
Result type vs Input type matchings are validated in these [tests](https://github.com/apache/spark/pull/38419/files#diff-d9ba3f61e2d32dabcdff1b0aeb78c40628efcea440eadcf13336a36d1a626282R1065)

Reply:
@srielau , I've updated the code to match and added test cases in sql/core/src/test/resources/sql-tests/inputs/trunc.sql
Ref : sql/core/src/test/resources/sql-tests/results/trunc.sql.out
struct<truncnumber(CAST(123456789.123456789 AS DECIMAL(18,9)), 8):decimal(18,9)>